src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -357,15 +356,18 @@ private async Task UpdateAdvertisingManifestAsync(ManifestId manifestId, bool in
                 {
                     File.Delete(packagePath);
                 }
-
-                var versionDir = Path.GetDirectoryName(packagePath);
-                if (Directory.Exists(versionDir) && !Directory.GetFileSystemEntries(versionDir).Any())
+                if (offlineCache == null || !offlineCache.HasValue)
```

Comment:
This would be better to guard against cases where `packagePath` is null:

```suggestion
                if (!string.IsNullOrEmpty(packagePath) && (offlineCache == null || !offlineCache.HasValue))
```